graph/src/components/store/entity_cache.rs

```diff
@@ -203,18 +202,111 @@ impl EntityCache {
 
         let query = DerivedEntityQuery {
             entity_type: EntityType::new(base_type.to_string()),
-            entity_field: field.name.clone().to_snake_case().into(),
+            entity_field: field.name.clone().into(),
             value: eref.entity_id.clone(),
             causality_region: eref.causality_region,
             id_is_bytes: id_is_bytes,
         };
 
-        let entities = self.store.get_derived(&query)?;
-        entities.iter().for_each(|(key, e)| {
-            self.current.insert(key.clone(), Some(e.clone()));
-        });
-        let entities: Vec<Entity> = entities.values().cloned().collect();
-        Ok(entities)
+        let mut entity_map = self.store.get_derived(&query)?;
+
+        for (key, entity) in entity_map.iter() {
+            // Only insert to the cache if it's not already there
+            // This is to avoid overwriting updates
```

Comment:
Since `self.current` only contains entities that have already been written to the store, there's no danger of clobbering an update. It's still good to add those entities to `self.current`, it's just that the comment is misleading.

Reply:
in that case should we just insert it without checking ? Is there a reason to keep the check?